### DISCUSSION TITLE
2024-08-08 회의록

### DISCUSSION THREAD

**Discussion by eunohhh** [nbc-final-a10/TripBuddy#215] [회의록]
### 🔥 MVP/UT 후 2주간 이어서 해결해야 할 내용들
- 남은 기능들(찜하기, 좋아요, 팔로우, 알림)
  - 스토리 / 찜하기 / 좋아요 - 오은 
  - 리치텍스트에디터 / 글쓰기 수정(*) / 글쓰기 삭제 / 팔로우 / 온도지수 로직 - 김병준
  - 알림(팔로우찜하기되는대로) - 김민지
  - 검색 - 곽현정
- 리팩토링 / 컨벤션
  - 탠스택쿼리로 전환(인밸리데이트쿼리)
- 디렉토리(컴포넌트) 구조 정리
- 데스크탑 버전
- 8/12 월요일에 전체 코드 리뷰 및 개선사항 논의

_(*)는 추가 논의가 필요한, 매우 하드한 작업_

### 🔥 차후 일정 점검(projects)
- 8/7 저녁 UT 회의 후 8/8(모레) 아침 회의에서 집중적으로 논의

### 🔥 티켓 일정 재분배
- 검색 어떻게 할지(단순검색 / 옵션검색  =>> 피드백 사항 : 피드백은 다 선택해야 된다고 생각하신 것 같은데, 실제로는 옵션이니 UI 에서 이런점을 보여주기)
  - 일단 >> 현행유지
- Image with src "/images/tuto0.webp" has legacy prop "layout". Did you forget to run the codemod?(튜토리얼)

### 🔥 실패했을 경우의 화면?
- **무언가 오류 error.tsx / not-found.tsx**
- 병준님은 우선 얼러트로

### 🔥 추가 논의가 필요한 사항
- 알림 아이콘 위치
- 글쓰기 아이콘 위치

### 🔥 결정 사항
- (데스크탑) 세로스크롤 가능하게하고, 화살표로 목록 이동하고, 좌우스크롤은 옵션

### 🔥 티켓 재분배
- 일정 체크 및 티켓 재분배 완료
- 8/15 목표로 달리기~
- 8/12 월요일 코드 리뷰 및 UT 당첨자 선정

**Comment by Captain-Kim** [comment]
## 김병준

- 내 마이페이지에 들어왔을 때 `프로필 수정` 버튼으로 보여야 하는데 `팔로우 하기` 버튼으로 잘못 렌더링 되는 버그가 잠깐 식별 됨.
  - -> 그러나 언제 그런 문제가 발생하는지 파악을 못함. 간헐적으로 발생하는 것 같은데 데이터가 캐싱되어서 그런가 하고 계속 테스트해보는데 재현이 안 됨.
  - -> 이렇게 된 경우 팔로우하기를 누르면 내가 나를 팔로우하는 게 가능한 이상한 현상 발생 -> 일단 이거라도 유효성 검사로 막아야 할 듯.

**Comment by slsl2** [comment]
## 김민지
- 알림 아이콘이 현재 trips 페이지에만 있는데 다른 페이지에서도 보이도록 하면 좋을 것 같습니다. 어떤 페이지에서 보이게 할지 정해야 함.
- contract 테이블에 contract_last_read_message_id 컬럼 추가해야 함
- getTimeSinceUpload.ts를 알림목록, 채팅목록 페이지에서도 재사용하면 좋을 것 같습니다. 여기선 '업로드'를 빼고 앞부분만 필요한데, '업로드' 글자를 getTimeSinceUpload.ts에서는 빼고 필요한 컴포넌트에서만 추가하는 방식으로 하는 건 어떨지 의견 여쭙고 싶습니다!
<image src="https://github.com/user-attachments/assets/8fc61dee-16af-4391-b529-70d2d0886513" width="300" />

**Reply by Captain-Kim** [reply]
`contract` 테이블에 `contract_last_read_message_id` 컬럼 추가 후 알려주세요. supabase types 추가하겠습니다.
nullable 해야 합니다.
